pyquil/tests/test_parser.py

```diff
@@ -50,6 +52,20 @@ def test_def_gate():
     _test(defgates, sqrt_x, hadamard)
 
 
+def test_def_gate_with_variables():
+    # Note that technically the RX gate includes -i instead of just i but this messes a bit with the test since
+    # it's not smart enough to figure out that -1*i == -i
+    theta = Parameter('theta')
+    rx = np.array([[cos(theta / 2), 1j * sin(theta / 2)],
+                   [1j * sin(theta / 2), cos(theta / 2)]])
+
+    defgate = 'DEFGATE RX(%theta):\n' \
+              '    cos(%theta/2), i*sin(%theta/2)\n' \
+              '    i*sin(%theta/2), cos(%theta/2)\n\n'
+
+    _test(defgate, DefGate('RX', rx, [theta]))
```

Comment:
AFAIU `_test` checks if the parsed `defgate` agrees with the `DefGate(...)` object, would it make sense to also test that `DefGate(...)` within a`Program` has the correct `out()`-put?

Reply:
Yep! I added that to test_quil